app/src/main/java/deakin/gopher/guardian/view/general/TaskAddActivity.kt

```diff
@@ -0,0 +1,122 @@
+package deakin.gopher.guardian.view.general
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.GravityCompat
+import androidx.drawerlayout.widget.DrawerLayout
+import com.google.firebase.database.FirebaseDatabase
+import deakin.gopher.guardian.R
+import deakin.gopher.guardian.model.GP
+import deakin.gopher.guardian.model.NextOfKin
+import deakin.gopher.guardian.model.Patient
+import deakin.gopher.guardian.model.Task
+import deakin.gopher.guardian.util.DataListener
+
+
+class TaskAddActivity : AppCompatActivity(), DataListener {
+    private lateinit var taskDescriptionEditText: EditText
+    private lateinit var patientIdEditText: EditText
+    private var patient: Patient? = null
+    private var task: Task? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_task_add)
+
+        taskDescriptionEditText = findViewById(R.id.taskDescriptionEditText)
+        patientIdEditText = findViewById(R.id.taskPatientIdEditText)
+
+        val submitButton : Button = findViewById(R.id.newTaskSubmitButton)
+        submitButton.setOnClickListener {
+            showSaveDialog()
+        }
+
+
+        val customHeader: CustomHeader = findViewById(R.id.taskCustomHeader)
+        val drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
```

Comment:
For the type casting, instead of `<>` can you use `val drawerLayout: DrawerLayout`? this is just for consistency because in other Kotlin files we are casting this way.

Reply:
AFAIK there's not real difference between these two styles. But casting with `:` is slightly quicker to read when reading left-to-right